src/js/common/schemaform/review/ReviewPage.jsx

```diff
@@ -39,194 +20,57 @@ const scrollToTop = () => {
-          event: `${formConfig.trackingPrefix}-validation-failed`,
-        });
-        Raven.captureMessage('Validation issue not displayed', {
-          extra: {
-            errors,
-            prefix: formConfig.trackingPrefix
-          }
-        });
-        this.props.setSubmission('status', 'validationError');
-      }
-      this.props.setSubmission('hasAttemptedSubmit', true);
-    }
-  }
-
-  handleEdit = (pageKey, editing, index = null) => {
-    const fullPageKey = `${pageKey}${index === null ? '' : index}`;
-    if (editing) {
-      this.setPagesViewed([fullPageKey]);
-    }
-    this.props.setEditMode(pageKey, editing, index);
-  }
-
-  handleToggleChapter(toggledChapter) {
-    if (this.props.openChapters.includes(toggledChapter)) {
-      this.props.closeReviewChapter(toggledChapter);
-    } else {
-      this.props.openReviewChapter(toggledChapter);
-    }
-  }
-
   render() {
-    const { route, form, contentAfterButtons, renderErrorMessage, formContext } = this.props;
-    const formConfig = route.formConfig;
-    const chapters = getActiveChapters(formConfig, form.data);
+    const {
+      formConfig,
+      pageList,
+      path
+    } = this.props;
 
     return (
       <div>
-        <div className="input-section">
-          <div>
-            {chapters.map(chapter => (
-              <ReviewCollapsibleChapter
-                key={chapter}
-                onEdit={this.handleEdit}
-                toggleButtonClicked={() => this.handleToggleChapter(chapter)}
-                open={this.props.openChapters.includes(chapter)}
-                pages={this.pagesByChapter[chapter]}
-                chapterKey={chapter}
-                setData={this.props.setData}
-                setValid={this.props.setValid}
-                uploadFile={this.props.uploadFile}
-                chapter={formConfig.chapters[chapter]}
-                viewedPages={this.state.viewedPages}
-                setPagesViewed={this.setPagesViewed}
-                formContext={formContext}
-                form={form}/>
-            ))}
-          </div>
-        </div>
-        <p><strong>Note:</strong> According to federal law, there are criminal penalties, including a fine and/or imprisonment for up to 5 years, for withholding information or for providing incorrect information. (See 18 U.S.C. 1001)</p>
-        <PrivacyAgreement required
-          onChange={this.props.setPrivacyAgreement}
-          checked={form.data.privacyAgreementAccepted}
-          showError={form.submission.hasAttemptedSubmit}/>
-        <SubmitButtons
-          onBack={this.goBack}
-          onSubmit={this.handleSubmit}
-          submission={form.submission}
-          renderErrorMessage={renderErrorMessage}/>
-        {contentAfterButtons}
+        <ReviewChapters
+          formConfig={formConfig}
+          pageList={pageList}/>
+        <SubmitController
+          formConfig={formConfig}
+          pageList={pageList}
+          path={path}/>
```

Comment:
`ReviewPage` composes two connected components. The only props it's passing are from its `ownProps` i.e. things not on the redux state.